test/contractBridge/whitelist.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2020 ChainSafe Systems
+ * SPDX-License-Identifier: LGPL-3.0-only
+ */
+
+const TruffleAssert = require('truffle-assertions');
+
+const BridgeContract = artifacts.require("Bridge");
+const TestWhitelistContract = artifacts.require("TestWhitelist");
+
+contract('Bridge - [whitelist]', async accounts => {
+    const chainID = 1;
+    const initialRelayers = accounts.slice(0, 3);
+    const initialRelayerThreshold = 2;
+
+    const adminAccount = accounts[0]
+    const genericAccount = accounts[3]
+    const testAccount = accounts[4]
+
+    let BridgeInstance;
+    let ADMIN_ROLE;
+    let WHITELISTER_ROLE;
+
+    beforeEach(async () => {
+        BridgeInstance = await BridgeContract.new(chainID, initialRelayers, initialRelayerThreshold, 0, 100);
+        ADMIN_ROLE = await BridgeInstance.DEFAULT_ADMIN_ROLE()
+        WHITELISTER_ROLE = await BridgeInstance.WHITELISTER_ROLE()
+    });
+
+    // Testing whitelist methods
+
+    it('should enable/disable whitelist and emit events', async () => {
+        assert.isFalse(await BridgeInstance.isWhitelistEnabled());
+        const enableTx = await BridgeInstance.enableWhitelist();
+        TruffleAssert.eventEmitted(enableTx, 'WhitelistEnabled');
+        assert.isTrue(await BridgeInstance.isWhitelistEnabled());
+        const disableTx = await BridgeInstance.disableWhitelist();
+        TruffleAssert.eventEmitted(disableTx, 'WhitelistDisabled');
+        assert.isFalse(await BridgeInstance.isWhitelistEnabled());
+    });
+
+    it('should add/remove whitelisted address and emit events', async () => {
+        assert.isFalse(await BridgeInstance.isWhitelisted(testAccount));
```

Comment:
Split this test into 2. Each unit test should check only 1 thing.

Reply:
[done](https://github.com/Cerebellum-Network/chainbridge-solidity/pull/4/files#diff-a62c2a6bda93583bed6bafc173326ced2d611a6ae8a37740d97001a613dceefeR49)